### ISSUE TITLE
[0.12.0] Panic in validateSeriesAndFields

### ISSUE THREAD

**Issue by toddboom** [influxdata/influxdb#6276]
While doing concurrent writes, this panic was encountered. It's reproducible with the following script:
https://gist.github.com/rw/63b75dbcb976e04f9f51ab973e6b79d9

This is on version `0.12.0` installed from Homebrew.

```
[query] 2016/04/10 15:20:01 CREATE DATABASE crash_db
[http] 2016/04/10 15:20:01 ::1 - - [10/Apr/2016:15:20:01 -0700] GET /query?q=CREATE+DATABASE+crash_db HTTP/1.1 200 40 - Go-http-client/1.1 5e623203-ff6a-11e5-8001-000000000000 1.340057ms
[tsm1wal] 2016/04/10 15:20:01 tsm1 WAL starting with 10485760 segment size
[tsm1wal] 2016/04/10 15:20:01 tsm1 WAL writing to /Users/rw/.influxdb/wal/crash_db/default/1
[tsm1] 2016/04/10 15:20:01 /Users/rw/.influxdb/data/crash_db/default/1 database index loaded in 1.609µs
fatal error: concurrent map read and map write

goroutine 59 [running]:
runtime.throw(0x8e01c0, 0x21)
        /usr/local/Cellar/go/1.6/libexec/src/runtime/panic.go:530 +0x90 fp=0xc8201e58e8 sp=0xc8201e58d0
runtime.mapaccess1_fast64(0x5ff940, 0xc8201f3800, 0x1, 0xc8200892c0)
        /usr/local/Cellar/go/1.6/libexec/src/runtime/hashmap_fast.go:112 +0x5a fp=0xc8201e5908 sp=0xc8201e58e8
github.com/influxdata/influxdb/tsdb.(*Shard).validateSeriesAndFields(0xc82016aa20, 0xc820430000, 0x320, 0x400, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /private/tmp/influxdb20160410-3418-xhad7n/influxdb-0.12.0/src/github.com/influxdata/influxdb/tsdb/shard.go:374 +0x1294 fp=0xc8201e5bc8 sp=0xc8201e5908
github.com/influxdata/influxdb/tsdb.(*Shard).WritePoints(0xc82016aa20, 0xc820430000, 0x320, 0x400, 0x0, 0x0)
        /private/tmp/influxdb20160410-3418-xhad7n/influxdb-0.12.0/src/github.com/influxdata/influxdb/tsdb/shard.go:237 +0xe3 fp=0xc8201e5de8 sp=0xc8201e5bc8
github.com/influxdata/influxdb/tsdb.(*Store).WriteToShard(0xc8200e8540, 0x1, 0xc820430000, 0x320, 0x400, 0x0, 0x0)
        /private/tmp/influxdb20160410-3418-xhad7n/influxdb-0.12.0/src/github.com/influxdata/influxdb/tsdb/store.go:632 +0x14e fp=0xc8201e5e30 sp=0xc8201e5de8
github.com/influxdata/influxdb/cluster.(*PointsWriter).writeToShard(0xc8200a9ea0, 0xc820436ca0, 0xc820156c6f, 0x8, 0x82fb28, 0x7, 0xc820430000, 0x320, 0x400, 0x0, ...)
        /private/tmp/influxdb20160410-3418-xhad7n/influxdb-0.12.0/src/github.com/influxdata/influxdb/cluster/points_writer.go:249 +0xd0 fp=0xc8201e5ee0 sp=0xc8201e5e30
github.com/influxdata/influxdb/cluster.(*PointsWriter).WritePoints.func1(0xc82017ee40, 0xc8200a9ea0, 0xc820436ca0, 0xc820156c6f, 0x8, 0x82fb28, 0x7, 0xc820430000, 0x320, 0x400)
        /private/tmp/influxdb20160410-3418-xhad7n/influxdb-0.12.0/src/github.com/influxdata/influxdb/cluster/points_writer.go:212 +0x8d fp=0xc8201e5f50 sp=0xc8201e5ee0
runtime.goexit()
        /usr/local/Cellar/go/1.6/libexec/src/runtime/asm_amd64.s:1998 +0x1 fp=0xc8201e5f58 sp=0xc8201e5f50
created by github.com/influxdata/influxdb/cluster.(*PointsWriter).WritePoints
        /private/tmp/influxdb20160410-3418-xhad7n/influxdb-0.12.0/src/github.com/influxdata/influxdb/cluster/points_writer.go:213 +0x411
```

**Comment by e-dard** [comment]
Dupe of #6235. Fixed as of `0.12.1`

**Comment by toddboom** [comment]
:+1: Thanks!
